core/src/main/java/org/infinispan/configuration/parsing/ParserRegistry.java

```diff
@@ -239,25 +246,21 @@ private boolean isSupportedNamespaceVersion(Namespace namespace, String version)
     * @param os the output stream where the configuration should be serialized to
     * @param globalConfiguration the global configuration. Can be null
     * @param configurations a map of named configurations
-    * @throws XMLStreamException
     */
-   public void serialize(OutputStream os, GlobalConfiguration globalConfiguration, Map<String, Configuration> configurations) throws XMLStreamException {
-      BufferedOutputStream output = new BufferedOutputStream(os);
-      XMLStreamWriter subWriter = XMLOutputFactory.newInstance().createXMLStreamWriter(output);
-      XMLExtendedStreamWriter writer = new XMLExtendedStreamWriterImpl(subWriter);
+   public void serialize(OutputStream os, GlobalConfiguration globalConfiguration, Map<String, Configuration> configurations) {
+      ConfigurationWriter writer = ConfigurationWriter.createConfigurationWriter(os);
       serialize(writer, globalConfiguration, configurations);
-      subWriter.close();
+      Util.close(writer);
```

Comment:
Should we be ignoring these exceptions?

Throwing `IOException` is more explicit and we already have the catch logic in place as previously `XMLStreamException` was thrown.